hnn_core/tests/test_drives.py

```diff
@@ -217,6 +218,7 @@ def test_add_drives():
             assert num_connections == \
                 np.around(len(net.gid_ranges[cell_type]) *
                           probability[cell_type]).astype(int)
+    simulate_dipole(net, tstop=1)  # smoke test
```

Comment:
This is a monolithic unit test - is there any way we can make it a little more targeted? Maybe we could put a test in `test_network.py` instead since it's more about how `NetworkBuilder` establishes connections.

Reply:
could you do instead `net._instantiate_drives(tstop=1)` ? Would that also do the test?

Reply:
I agree with your point @rythorpe, but considering it only runs once, it's a relatively minimal impact on the run length of tests. 

@jasmainak that works as well and addresses the issue (that I am aware of).

In my view, the reason this bug exists in the first place is that the drives API was only tested in pieces, but broke when everything was put together. I worry that this will come up again since we are planning to heavily change the API in the future. Obviously we can't run every permutation of a network through `simulate_dipole()` as a test, but how do we avoid this in the future?

Let me know if you agree or not and I'll update the line accordingly.

Reply:
Sorry I forgot to respond to this. I agree that using the public API as much as possible for tests is always the best strategy

Reply:
Apologies for the late reply @ntolley. I agree that we need better tests that clearly test for correct implementation of the API, however, I have also found it helpful when unit tests can help diagnose bugs for new contributors. There's a lot of very specific and fragile code contained in `NetworkBuilder._gid_assign()` and `NetworkBuilder._create_cells_and_drives` that helps instantiate the python and nrn versions of the network. Since I was one of the last ones to add to the mess prior to this PR, I just think we would be well served by differentiating between tests that identify mismatch between what the user enters through the API and what `Network` produces, and tests that identify mismatch between `Network` and `NetworkBuilder` objects, especially when it comes to cell instantiation and connectivity. All that said, there are numerous ways to accomplish this and I'm guessing that your solution here will probably serve us the best until we can clean and shore up some of the drive tests.